cmd/infra/aws/create_cli_role.go

```diff
@@ -0,0 +1,246 @@
+					"route53:ListHostedZones",
+					"route53:ChangeResourceRecordSets",
+					"route53:ListResourceRecordSets",
+					"route53:DeleteHostedZone",
+					"route53:AssociateVPCWithHostedZone",
+					"route53:ListHostedZonesByName"
+				],
+				"Resource": "*"
+			},
+			{
+				"Sid": "S3",
+				"Effect": "Allow",
+				"Action": [
+					"s3:ListAllMyBuckets",
+					"s3:ListBucket",
+					"s3:DeleteObject",
+					"s3:DeleteBucket"
+				],
+				"Resource": "*"
+			}
+		]
+	}`
+)
+
+type CreateCLIRoleOptions struct {
+	AWSCredentialsFile string
+	RoleName           string
+	AdditionalTags     map[string]string
+}
+
+func NewCreateCLIRoleCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "cli-role",
+		Short:        "Creates AWS IAM role for the CLI to assume",
+		SilenceUsage: true,
+	}
+
+	opts := CreateCLIRoleOptions{
+		AWSCredentialsFile: "",
+		RoleName:           "hypershift-cli-role",
+	}
+
+	cmd.Flags().StringVar(&opts.AWSCredentialsFile, "aws-creds", opts.AWSCredentialsFile, "Path to an AWS credentials file (required)")
+	cmd.Flags().StringVar(&opts.RoleName, "name", opts.RoleName, "Role name")
+	cmd.Flags().StringToStringVarP(&opts.AdditionalTags, "additional-tags", "t", opts.AdditionalTags, "Additional tags to apply to the role created (e.g. 'key1=value1,key2=value2')")
+
+	cmd.MarkFlagRequired("aws-creds")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := opts.Run(cmd.Context()); err != nil {
+			log.Log.Error(err, "failed to create cli role")
+			return err
+		}
+		return nil
+	}
+
+	return cmd
+}
+
+func (o *CreateCLIRoleOptions) Run(ctx context.Context) error {
+	tags, err := o.ParseAdditionalTags()
+	if err != nil {
+		return err
+	}
+
+	awsSession := awsutil.NewSession("cli-create-role", o.AWSCredentialsFile, "", "", "")
+	awsConfig := awsutil.NewConfig()
+	iamClient := iam.New(awsSession, awsConfig)
+	stsClient := sts.New(awsSession, awsConfig)
+
+	trustPolicy, err := assumeRoleTrustPolicy(ctx, stsClient)
+	if err != nil {
+		return err
+	}
+
+	createIAMRoleOpts := CreateIAMRoleOptions{
+		RoleName:          o.RoleName,
+		TrustPolicy:       trustPolicy,
+		PermissionsPolicy: cliRolePolicy,
+		additionalIAMTags: tags,
+	}
+
+	roleArn, err := createIAMRoleOpts.CreateRoleWithInlinePolicy(ctx, iamClient)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Sucessfully created/updated role %s, arn: %s\n", o.RoleName, roleArn)
+	return nil
+}
+
+func assumeRoleTrustPolicy(ctx context.Context, client *sts.STS) (string, error) {
+	identity, err := client.GetCallerIdentityWithContext(ctx, &sts.GetCallerIdentityInput{})
+	if err != nil {
+		return "", err
+	}
+
+	assumeRolePolicy := fmt.Sprintf(`{
+		"Version": "2012-10-17",
+		"Statement": [
+			{
+				"Effect": "Allow",
+				"Principal": {
+					"AWS": "%s"
+				},
+				"Action": "sts:AssumeRole"
+			}
+		]
+	}`, *identity.Arn)
+
+	return assumeRolePolicy, nil
+}
+
+func (o *CreateCLIRoleOptions) ParseAdditionalTags() ([]*iam.Tag, error) {
```

Comment:
This looks like a copy-pasta from `iam.go` - please DRY it out

Reply:
same as https://github.com/openshift/hypershift/pull/4027#discussion_r1599624238